src/DataCollection.Shared/ViewModels/AttachmentsViewModel.cs

```diff
@@ -110,14 +110,16 @@ public ICommand OpenAttachmentCommand
                                 await attachment.LoadAsync();
                             }
 
+                            // HACK: This workflow is in place until API changes occur to save the file name with its proper extension
+
                             // create file name from runtime attachment filename hash and the actual attachment name
                             var runtimeFileHash = attachment.Filename.Split('.').FirstOrDefault();
                             var newFileName = runtimeFileHash + "." + attachment.Name;
```

Comment:
I don't like the behavior of copying the file in this case because it can clutter the user's files. Could the file be copied to a temp directory prior to opening, rather than the original file location?

Reply:
@nCastle1 I updated the workflow to use a temp directory when the file cannot be opened.  That way the attachment maintains the same name, which should make it less confusing for the user. This is the same workflow that will be implemented later in the API to match what iOS and Android are doing. See internal issue 7249 for details.